### DISCUSSION TITLE
Unable to import prisma model from db in mutation

### DISCUSSION THREAD

**Discussion by spidertwin2** [blitz-js/blitz#2430] [Q&A] [answered]
Hi,

My schema has a Location model, but I am unable to use its types in a mutation.

schema.prisma
```
model Location {
  id       Int       @id @default(autoincrement())
  name     String
  label    String    @unique
  locality String?
  region   String?
  country  String?
  //one location can be shared by many profiles
  profiles Profile[]
}
```
updateLocation.ts

```
import db, { Location } from "db"  //error

type LocationProps = Omit<Location, "id">

export default async function updateLocation(location: LocationProps, ctx: Ctx) {
  ctx.session.$authorize

  if (!ctx.session.userId) return null

  try {
    // perform mutation
  } catch (error) {
    console.log("Error while updating your location: ", error)
  }
}
```
console error -

```
Type error: Module '"../../../../db"' has no exported member 'Location'. 
Did you mean to use 'import Location from "../../../../db"' instead?
```

Thanks :green_apple:

**Comment by flybayer** [comment]
Usually you need to run `blitz prisma migrate dev` after updating the schema to (1) migrate the DB and (2) generate updated Prisma client. Did you do that?

If so, then try running `blitz prisma generate` which will manually generate a new prisma client so the types will be updated.

**Reply by spidertwin2** [reply]
I am able to use `updateLocation` mutation with `blitz dev`.

`blitz prisma generate`

````
✔ Generated Prisma Client (2.22.1) to ./node_modules/@prisma/client in 217ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
````
`blitz prisma migrate dev`
````
Already in sync, no schema change or pending migration was found.

✔ Generated Prisma Client (2.22.1) to ./node_modules/@prisma/client in 235ms
````

`railway up`

```
Failed to compile.

./app/account/mutations/updateLocation.ts:2:14
Type error: Module '"../../../../db"' has no exported member 'Location'. 
Did you mean to use 'import Location from "../../../../db"' instead?

  1 | import { Ctx } from "blitz"
> 2 | import db, { Location } from "db"
    |              ^
  3 | 
  4 | type LocationProps = Omit<Location, "id">
  5 | 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! myapp@1.0.0 build: `blitz build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the myapp@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Reply by flybayer** [reply]
Oh, it's a Railway issue. You may need to add `blitz prisma generate` to your railway build script

**Comment by spidertwin2** [comment, marked as answer]
I updated my build script to `yarn migrate:dev && blitz build`.  

But a `TypeError: outputDir.endsWith is not a function` error was thrown.  It was caused by version mismatch of `@prisma/client` and `prisma`, which was resolved by updating both packages to the latest version `2.24.0`.

Thanks @coffee-cup!
:coffee:

**Reply by flybayer** [reply]
cool!

P.S. Not sure what your `migrate:dev` script does, but make sure you aren't running `blitz prisma migrate dev` for your production DB. Use `blitz prisma migrate deploy` instead. Also you want to run your migrations *after* `blitz build` so migrations only run on build success

**Reply by spidertwin2** [reply]
sorry, I don't understand..here are my scripts, please let me know if anything is amiss.

```
"scripts": {
    "dev": "blitz dev",
    "build": "yarn migrate:dev && blitz build",
    "start": "blitz start --port ${PORT-3000}",
    "studio": "blitz prisma studio",
    "start:production": "blitz start --production --port ${PORT-3000}",
    "migrate:dev": "blitz prisma migrate dev --preview-feature",
    "migrate:deploy": "blitz prisma migrate deploy --preview-feature",
    "lint": "eslint --ignore-path .gitignore --ext .js,.ts,.tsx .",
    "test": "jest",
    "test:watch": "jest --watch"
  },
```

**Reply by coffee-cup** [reply]
I think the build script should be

```
"build": "blitz prisma generate && blitz build && yarn migrate:deploy"
```

So that
- Prisma type files are generated
- Blitz app is built
- DB is migrated (only if the build passes)
